aptos-move/aptos-release-builder/src/components/feature_flags.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::utils::*;
+use anyhow::Result;
+use aptos_types::on_chain_config::FeatureFlag;
+use move_model::{code_writer::CodeWriter, emit, emitln, model::Loc};
+
+pub struct Features {
+    pub enabled: Vec<FeatureFlag>,
+    pub disabled: Vec<FeatureFlag>,
+}
+
+pub fn generate_feature_upgrade_proposal(
```

Comment:
Here and for other components - I wonder if we should add some logic that checks what the current network state is before generating scripts. If there's no diff for feature flags for example, we shouldn't generate any.

Reply:
I'll add this logic in future PR.